docs/getting_started/java/first_program_in_java/index.md

```diff
@@ -1,373 +1,757 @@
+
+:::note Retryable Errors
+
+Although not shown in this code sample, you can specify Activity errors that Temporal should not attempt to retry.
+Set the Retry Option's non-retryable errors field.
+This provides you with the flexibility to choose the circumstances to retry.
+
+:::
+
+Now that you've seen the code, it's time to look at the Workflow in action.
+In the next sections, you'll launch the Temporal Server and create a Workflow Execution.
+
+## Start the Workflow
+
+In this section, you'll run and observe a Workflow using the project code and the Temporal Web UI.
+
 
-<!--SNIPSTART money-transfer-project-template-java-shared-constants-->
-[src/main/java/moneytransferapp/Shared.java](https://github.com/temporalio/money-transfer-project-template-java/blob/4e1ca12d8d1c058c67d810efe8dcf68f7b318208/src/main/java/moneytransferapp/Shared.java)
-```java
-public interface Shared {
+### Prepare to run
 
-    static final String MONEY_TRANSFER_TASK_QUEUE = "MONEY_TRANSFER_TASK_QUEUE";
-}
+Start running a [Temporal Service](https://docs.temporal.io/clusters) by launching it from a Terminal session, as discussed in the [Getting Started tutorial](https://learn.temporal.io/getting_started/java/dev_environment/):
+
+
+```bash
+temporal server start-dev
 ```
+
+If a Service is already running, you do not (and cannot) have to re-launch it again.
+If you are running a Unix-like system, you can check for this by issuing `pgrep`:
+
+
+```bash
+pgrep -fl temporal
+```
+
+### Workflow Execution
+
+When you _start_ a Workflow, you tell the Temporal Server, "Track the state of a Workflow that uses this method signature."
+This tracked state is called a Workflow Execution, and follows the code from the Workflow and Activity definitions you just saw through their lifetimes.
+The Workflow Execution represents a complete history of the Workflow's lifetime, including its intermediate states, its retries, and so forth.
+
+The Service breaks down the Workflow Execution into pieces, called Tasks.
+There are Workflow Tasks and Activity Tasks.
+Workers look for those Tasks on "Task Queues" (called "polling" for Tasks).
+That's why when you set up a Java Worker, you must tell it the Task Queue it should use to look for work.
+
+A Worker runs each piece of the Workflow separately, Task by Task.
+Whether that Task runs perfectly or it encounters issues, the Worker relays the execution events and results back to the Server.
+This is recorded into the Event History for the Workflow Execution.
+
+Workflow Executions "live" on the Temporal Server.
+In the most basic form, a Workflow Execution consists of a set of set-up data and an Event history.
+You initiate a Workflow Execution with the name of a Workflow Definition (the `MoneyTransferWorkflow` class), a Task Queue name to use, and optional input data.
+Each Workflow Execution has an identifier.
+If you do not supply one, the Service creates a random unique ID to use instead.
```

Comment:
This is actually only true for certain SDKs. In some SDKs you _must_ provide one. Might want to double check this.

Reply:
Adding it to my check list too. Thanks again.